microcosm/loaders/__init__.py

```diff
@@ -32,5 +33,5 @@ def load_from_dict(dct=None, **kwargs):
     dct.update(kwargs)
 
     def _load_from_dict(metadata):
-        return dict(dct)
+        return Configuration(dct)
```

Comment:
This line was incorrect.  Loaders are supposed to return `Configuration` objects.

This change makes me a little bit nervous due to the expansive scope of downstream repos using the `load_from_dict` function, but `Configuration` subclasses `dict`, and it passes all unit tests here, so it should work fine.

Reply:
I mean according to the comment above: 
"A configuration loader is any function that accepts `Metadata` and
returns a `dict` (or `Configuration` model)." So I think either should be fine. I think using a Configuration model is only necessary when code internal to the loader needs the extra merging logic.